lib/bundler/remote_specification.rb

```diff
@@ -73,7 +73,15 @@ def to_s
     end
 
     def dependencies
-      @dependencies || method_missing(:dependencies)
+      @dependencies ||= begin
+        deps = method_missing(:dependencies)
+        unless deps.all? {|d| d.is_a?(Gem::Dependency) }
+          # allow us to handle when the specs dependencies are an array of array of string
+          # see https://github.com/bundler/bundler/issues/5797
+          deps.map! {|d| Gem::Dependency.new(*d) }
```

Comment:
This logic states that you map all deps to a G::D, but what happens if (for some unknown reason), we have `[G::D, ['',''], G::D]`. In this example, you'd init a G::D with a G::D - which might be ok (I'm not sure).

Since `deps.all` already iterates through everything, what if we collapsed to this:
```ruby
deps = deps.map { |d| d.is_a?(Gem::Depencency) ? d :  Gem::Dependency.new(*d) }
```